### PR TITLE
Wire contexts to Batch controllers

### PR DESCRIPTION
**Pull request by damemi**
#### What type of PR is this?
/kind design

#### What this PR does / why we need it:
This wires a context to Batch controllers in kube-controller-manager, so that KCM can eventually release on cancel.

This originally came from PR https://github.com/kubernetes/kubernetes/pull/101379, which builds on top of https://github.com/kubernetes/kubernetes/pull/101125

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Review by damemi — COMMENTED** [review]
/cc @soltysh 
test failure was a known flake

**Review by soltysh — CHANGES_REQUESTED** [review]

**Review by soltysh — APPROVED** [review]
/lgtm
/approve
/label tide/merge-method-squash

File: cmd/kube-controller-manager/app/batch.go [hunk 1]
```diff
@@ -34,18 +34,20 @@ func startJobController(ctx context.Context, controllerContext ControllerContext
 		controllerContext.InformerFactory.Core().V1().Pods(),
 		controllerContext.InformerFactory.Batch().V1().Jobs(),
 		controllerContext.ClientBuilder.ClientOrDie("job-controller"),
-	).Run(int(controllerContext.ComponentConfig.JobController.ConcurrentJobSyncs), ctx.Done())
+	).Run(ctx, int(controllerContext.ComponentConfig.JobController.ConcurrentJobSyncs))
 	return nil, true, nil
 }
 
 func startCronJobController(ctx context.Context, controllerContext ControllerContext) (controller.Interface, bool, error) {
+
```
Comment (sttts) [inline cmd/kube-controller-manager/app/batch.go:42]: nit: empty lines at the top are evil

**Pull request merged**